File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        
        if head == None: return None

        h = head 
        p = h
        p2 = h.next

        while p2 :
            if p.val == p2.val:
                p.next = p2.next
                p2 = p.next 
            else:
                p = p.next
                p2 = p.next

        return h

File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
    
    if head == nil{
        return nil
    }

    h := head
    p1 := h
    p2 := h.Next

    for p2 != nil {

        if p1.Val == p2.Val {
            p1.Next = p2.Next
            p2 = p1.Next
        }else{
            p1 = p1.Next
            p2 = p1.Next
        }
    }
    return h
}